inspirehep/modules/records/receivers.py

```diff
@@ -361,3 +361,21 @@ def check_if_record_is_going_to_be_deleted(sender, *args, **kwargs):
         else:
             record = get_db_record(collection, control_number)
             soft_delete_pidstore_for_record(record.id)
+
+
+@after_record_update.connect
+def check_if_record_has_merged(sender, *args, **kwargs):
```

Comment:
I think it could be more accurate to have the trigger of when records are merged by listening to changes in the `pidstore_redirect` table. (using SQLAlchemy events - http://docs.sqlalchemy.org/en/latest/orm/events.html#sqlalchemy.orm.events.MapperEvents.after_update)


```python
from sqlalchemy import event

# standard decorator style
@event.listens_for(SomeClass, 'after_update')
def receive_after_update(mapper, connection, target):
    "listen for the 'after_update' event"

    # ... (event handling logic) ...
```